bcipy/signal/model/evaluate/fusion.py

```diff
@@ -0,0 +1,370 @@
+                    temp))  # Delta_t = X_t - mu
+        centralized_data_dict[sym] = np.array(centralized_data_dict[sym])
+        time_average_per_symbol[sym] = np.mean(np.array(time_average_per_symbol[sym]), axis=0)
+
+    # Take the time average of the gaze data:
+    centralized_gaze_data = np.zeros_like(preprocessed_gaze_data)
+    for i, (_, sym) in enumerate(zip(preprocessed_gaze_data, target_symbols)):
+        centralized_gaze_data[i] = gaze_model.substract_mean(preprocessed_gaze_data[i], time_average_per_symbol[sym])
+
+    '''Use bootstrap resampling for both EEG and Gaze data'''
+    n_iterations = 1
+    eeg_acc = []
+    gaze_acc = []
+    fusion_acc = []
+    # selection length is the length of eeg or gaze data, whichever is smaller:
+    selection_length = min(len(eeg_inquiries[1]), len(preprocessed_gaze_data))
+    for iter in range(n_iterations):
+        # Pick a train and test dataset (that consists of non-train elements) until test dataset is not empty:
+        train_indices = resample(list(range(selection_length)), replace=True, n_samples=100)
+        test_indices = np.array([x for x in list(range(selection_length)) if x not in train_indices])
+        if len(test_indices) == 0:
+            break
+
+        train_data_eeg = eeg_inquiries[:, train_indices, :]
+        test_data_eeg = eeg_inquiries[:, test_indices, :]
+        eeg_inquiry_timing = np.array(eeg_inquiry_timing)
+        train_eeg_inquiry_timing = eeg_inquiry_timing[train_indices]
+        test_eeg_inquiry_timing = eeg_inquiry_timing[test_indices]
+        inquiry_symbols_test = np.array([])
+        for t_i in test_indices:
+            inquiry_symbols_test = np.append(inquiry_symbols_test, inquiry_symbols[t_i * 10:(t_i + 1) * 10])
+        inquiry_symbols_test = inquiry_symbols_test.tolist()
+
+        # Now extract the inquiries from trials for eeg model fitting:
+        preprocessed_train_eeg = eeg_model.reshaper.extract_trials(
+            train_data_eeg, trial_duration_samples, train_eeg_inquiry_timing)
+        preprocessed_test_eeg = eeg_model.reshaper.extract_trials(
+            test_data_eeg, trial_duration_samples, test_eeg_inquiry_timing)
+
+        # train and save the eeg model a pkl file
+        # Flatten the labels (0=nontarget/1=target) prior to model fitting
+        erp_train_labels = eeg_inquiry_labels[train_indices].flatten().tolist()
+        # erp_test_labels = eeg_inquiry_labels[test_indices].flatten().tolist()
+        eeg_model.fit(preprocessed_train_eeg, erp_train_labels)
+        eeg_model.metadata = SignalModelMetadata(device_spec=device_spec_eeg,
+                                                 transform=default_transform)
+        # save_model(eeg_model, Path(data_folder, f"model_{eeg_model.auc:0.4f}.pkl"))
+        preferences.signal_model_directory = data_folder
+
+        # extract train and test indices for gaze data:
+        centralized_gaze_data_train = centralized_gaze_data[train_indices]
```

Comment:
Should some of this live in the Gaze Model itself?

Reply:
`centralized_gaze_data` is the output of `subtract_mean` function which lies in `gaze_model`. Line 244 only picks a subset from that dataset (i.e. `train_indices`) for quick model training